packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -744,6 +720,101 @@ public ResponseEntity<TaskResponse> createGenerateResponseTask(
 		return ResponseEntity.ok().body(resp);
 	}
 
+	/**
+	 * This endpoint will dispatch a few GoLLM tasks to enrich model metadata,
+	 * including enriching the AMR and creating the model card
+	 * @param modelId UUID of the model to enrich
+	 * @param documentId UUID of the document to use for enrichment
+	 * @param mode TaskMode to run the task in (is this ASYNC?)
+	 * @param projectId UUID of the project to associate the task with for permissions
+	 * @param overwrite boolean to determine if the model should be overwritten
+	 * @return TaskResponse with the task ID
+	 */
+	@GetMapping("/enrich-model-metadata")
+	@Secured(Roles.USER)
+	@Operation(summary = "Dispatch a multiple GoLLM tasks to enrich model metadata")
+	@ApiResponses(
+		value = {
+			@ApiResponse(
+				responseCode = "200",
+				description = "Dispatched successfully",
+				content = @Content(
+					mediaType = "application/json",
+					schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = TaskResponse.class)
+				)
+			),
+			@ApiResponse(
+				responseCode = "404",
+				description = "The provided model or document arguments are not found",
+				content = @Content
+			),
+			@ApiResponse(responseCode = "500", description = "There was an issue dispatching the request", content = @Content)
+		}
+	)
+	public ResponseEntity<TaskResponse> createEnrichModelMetadataTask(
+		@RequestParam(name = "model-id", required = true) final UUID modelId,
+		@RequestParam(name = "document-id", required = false) final UUID documentId,
+		@RequestParam(name = "mode", required = false, defaultValue = "ASYNC") final TaskMode mode,
+		@RequestParam(name = "project-id", required = false) final UUID projectId,
+		@RequestParam(name = "overwrite", required = false, defaultValue = "false") final boolean overwrite
+	) {
+		final Schema.Permission permission = projectService.checkPermissionCanRead(
+			currentUserService.get().getId(),
+			projectId
+		);
+
+		// Grab the document
+		final Optional<DocumentAsset> document = documentAssetService.getAsset(documentId, permission);
+
+		// make sure there is text in the document. We don't need a document but if we do have one it can't be empty
+		if (document.isPresent() && (document.get().getText() == null || document.get().getText().isEmpty())) {
+			log.warn(String.format("Document %s has no extracted text", documentId));
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("document.extraction.not-done"));
+		}
+
+		// Grab the model
+		final Optional<Model> model = modelService.getAsset(modelId, permission);
+		if (model.isEmpty()) {
+			log.warn(String.format("Model %s not found", modelId));
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("model.not-found"));
+		}
+
+		final TaskRequest req;
+
+		if (document.isPresent()) {
```

Comment:
Do we need this condition with the test on [Line 770](https://github.com/DARPA-ASKEM/terarium/pull/5019/files#diff-45634443c497336d9e2e95ad796b7a09e0d7f9548c020d74a9497ef52759e636R770)?

Reply:
We do - the check on line 770 is checking to make sure that if there is a document it has content (which is a safety check that was there before so it feels important). However it is a valid case to have *no* document passed in, in which case we still need this.